Elysium/src/main/java/hse/elysium/serverspring/auth/JwtService.java

```diff
@@ -0,0 +1,74 @@
+package hse.elysium.serverspring.auth;
+
+import hse.elysium.entities.User;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.stereotype.Service;
+import io.jsonwebtoken.*;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class JwtService {
+
+    private static final String SECRET_KEY = "4A614E645267556B58703273357638792F413F4428472B4B6250655368566D59";
+
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public String extractIssuer(String token) {
+        return extractClaim(token, Claims::getIssuer);
+    }
+
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver) {
+        final Claims claims = extractAllClaims(token);
+        return claimsResolver.apply(claims);
+    }
+
+    public String generateToken(User user) {
+        return generateToken(new HashMap<>(), user);
```

Comment:
Why is a new map passed in?
Is `generateToken` called somewhere with an existing map?

Reply:
because jwt token requires extra claims